src/pkg/logger/logger.go

```diff
@@ -145,6 +154,36 @@ func New(cfg Config) (*slog.Logger, error) {
 	return log, nil
 }
 
+// defaultCtxKey provides a default key if one is not passed into From.
+var defaultCtxKey = CtxKey("logger")
+
+// From takes a context and reads out a "logger" value, optionally taking a key string. If multiple keys are provided,
+// any after the first will be ignored. Note that if From does not find a value, or that value is not a *slog.Logger,
+// it will return return nil.
+//
+// Usage:
+//
+//	l := From(ctx)
+//	l := From(ctx, "logger2")
+func From(ctx context.Context, key ...CtxKey) *slog.Logger {
```

Comment:
Most other logger implementations either as a from context or discard, or they return an error when the logger is not found. I think we should return a discard logger if it is not found in the context.

Reply:
I agree abt the discard logger - updated in latest